starknet/src/space/space.cairo

```diff
@@ -41,13 +84,35 @@ trait ISpace<TContractState> {
         metadata_uri: Array<felt252>,
         dao_uri: Array<felt252>,
     );
+    /// Creates a new proposal. Users must go through an authenticator before calling this function.
+    ///
+    /// # Arguments
+    ///
+    /// * `author` - The author of the proposal.
+    /// * `metadata_uri` - The metadata URI of the proposal. Could be a link to a discussion page, etc.
+    /// * `execution_strategy` - The execution strategy of the proposal. This is the contract that will be called
+    ///                          when the proposal is finalized.
+    /// * `user_proposal_validation_params` - The user proposal validation params. These are the params that will be
+    ///                                      passed to the proposal validation strategy.
     fn propose(
         ref self: TContractState,
         author: UserAddress,
         metadata_uri: Array<felt252>,
         execution_strategy: Strategy,
         user_proposal_validation_params: Array<felt252>,
     );
+
+    /// Casts a vote on a given proposal. Users must go through an authenticator before calling this function.
+    ///
+    /// # Arguments
+    ///
+    /// * `voter` - The voter.
+    /// * `proposal_id` - The proposal id to vote on.
+    /// * `choice` - The choice to vote for.
+    /// * `user_voting_strategies` - The voting strategies that the user wants to vote with. This allows to only 
+    ///                             query the `user_voting_strategies` for voting power instead of iterating through all
+    ///                             allowde voting strategies.
```

Comment:
The strategies to use to compute the voter's voting power, each consisting of a strategy index and an array of user provided parameters.

Reply:
6d158bf4a956c397eaea2bd5c89c2b3d44f824e7